p2p/peerexchange/bookchecker_test.go

```diff
@@ -154,6 +155,93 @@ func TestDiscovery_GetRandomPeers(t *testing.T) {
 	})
 }
 
+func TestHost_ReceiveAddressesOnCheck(t *testing.T) {
+	t.Parallel()
+
+	hosts, instances := setupOverNodes(t, 2)
+	nodeA, nodeB := instances[0], instances[1]
+
+	best, err := bestHostAddress(nodeA.host)
+	require.NoError(t, err)
+
+	connectedNodeAddress := addressFromHost(t, hosts[1])
+	nodeA.book.AddAddress(connectedNodeAddress, best)
+	wrapDiscovery(instances)
+
+	// check that after bootstrap, nodeB has only addresss of nodeA and nodeA has only addresss of nodeB
+	require.Eventually(t, func() bool {
+		aAddresses := nodeA.book.GetAddresses()
+		bAddresses := nodeB.book.GetAddresses()
+		if len(aAddresses) != 1 || len(bAddresses) != 1 {
+			return false
+		}
+		return aAddresses[0].ID.String() == nodeB.host.ID().String() && bAddresses[0].ID.String() == nodeA.host.ID().String()
+	}, 4*time.Second, 100*time.Millisecond, "nodeA should have only addresss of nodeB and nodeB should have only addresss of nodeA")
+
+	// disconnect nodeA and node. sleep 3 sec need for addresses become outdated and ready for check
+	require.NoError(t, nodeB.host.Network().ClosePeer(nodeA.host.ID()))
+	time.Sleep(3 * time.Second)
```

Comment:
rm

Reply:
this is need cause for get addresses, we watch at `address.LastUsage: time.Time` 
I cant modify address in addressBook, cause it in private struct. Only write special method, for tests, but it is not very good. Better just wait. This sleep not mean that some request will be executed, so it is not a crime here)

Reply:
yes, if find way to avoid sleep)